### PR TITLE
Quest 2 testing suite

### PR DESCRIPTION
**Pull request by JoeArmani**
Quest 2 automated testing suite

We needed a detailed automated testing suite here so we can safely work on the codebase when needed without encountering regressions or having to manually run all our surveys through testing.

This suite includes unit and Playwright tests, pinned to locked versions of our surveys, to address the cases and edge cases we've encountered over 2+ years of work on this system, with extra emphasis on survey grids, forid handling, piped variables, conditional evaluation, functions, statefulness, and tree handling.

This precedes upcoming event-handling and accessibility work that will very likely require DOM management and some structural changes, and should make it safer for developers to work in the repo without risking cascading bugs when making changes.

It also includes already-identified issues to resolve when working through https://github.com/episphere/connect/issues/1587 & https://github.com/episphere/connect/issues/1079

### CODE REVIEW

File: tests/e2e/support/test.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+  await page.route('**/*', async (route) => {
+    if (shouldUseLocalNetwork(route.request().url())) {
+      await route.continue();
+      return;
+    }
+
+    diagnostics.unexpectedRequests.push(route.request().url());
+    await route.abort('blockedbyclient');
+  });
```
[the remark this answers is not in the document]
Comment (JoeArmani) [reply tests/e2e/support/test.js:78]: No, Playwright invokes matching route handlers in reverse-registration order. The specific asset routes are registered after the catch-all. So they are fulfilled first. The catch-all handles only remaining external requests. Existing tests assert that the image and stylesheet requests were fulfilled locally.

File: vitest.config.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { defineConfig } from 'vitest/config';
+import { questDependencyResolver } from './tests/config/moduleResolution.js';
+
+export default defineConfig({
+  plugins: [questDependencyResolver],
+  test: {
+    environment: 'jsdom',
+    environmentOptions: {
+      jsdom: {
+        // Match the non-local participant host used by Connect. Tests that
```
[the remark this answers is not in the document]
Comment (JoeArmani) [reply vitest.config.js]: Updated

File: tests/integration/eventHandlers.spec.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+      key: 'Enter',
+      keyCode: 13,
+    });
+    expect(input.dispatchEvent(enter)).toBe(false);
+    expect(enter.defaultPrevented).toBe(true);
+
+    input.value = 'debounced value';
+    input.dispatchEvent(new InputEvent('input', { bubbles: true, data: 'e', inputType: 'insertText' }));
+    await new Promise((resolve) => setTimeout(resolve, 300));
+    expect(quest.state.getActiveQuestionState()).toEqual({});
```
[the remark this answers is not in the document]
Comment (JoeArmani) [reply tests/integration/eventHandlers.spec.js:38]: Updated

File: tests/corpus/lock.json [hunk 4]
```diff
@@ -0,0 +1,361 @@
+{
+  "$schema": "./schemas/lock.schema.json",
+  "schemaVersion": 1,
+  "repository": "episphere/questionnaire",
+  "repositoryUrl": "https://github.com/episphere/questionnaire.git",
+  "defaultBranch": "main",
+  "commit": "7ae99a22af325cf0e14be047a7636462db9bfd50",
```
Comment (anthonypetersen) [inline tests/corpus/lock.json:7]: will this need to change over time if existing markdowns are updated?

File: tests/corpus/scripts/fetchCorpus.mjs [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  return options;
+}
+
+function githubHeaders() {
+  const headers = {
+    Accept: 'application/vnd.github+json',
+    'X-GitHub-Api-Version': '2022-11-28',
+    'User-Agent': 'quest-corpus-fetcher',
+  };
+  const token = process.env.GITHUB_TOKEN || process.env.GH_TOKEN;
```
Comment (anthonypetersen) [inline tests/corpus/scripts/fetchCorpus.mjs]: will this need to be set manually by the developer running the suite?

File: tests/knownDefects/registry.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
+const DEFAULT_OWNER = 'Quest maintainers';
+const DEFAULT_EXPIRY = '2027-08-04';
```
Comment (anthonypetersen) [inline tests/knownDefects/registry.js:2]: This file is being referenced in tests/unit/knownDefectRegistry.spec.js, it looks like that will cause this to run in the blocking `quality` lane. On 08/05/27 will this cause future PRs to automatically fail automated testing?

**Review by anthonypetersen — APPROVED** [review]
Incredibly thorough!! This will be so nice to have and ease developers minds when they make changes in the future. A few small things that Claude pointed out that I felt were worth bringing up.

File: tests/corpus/lock.json (hunk 4 above)
Comment (JoeArmani) [reply tests/corpus/lock.json:7]: Yes, but only when we intentionally adopt a newer questionnaire snapshot. This keeps the testing expectations stable in the meantime.

File: tests/corpus/scripts/fetchCorpus.mjs (hunk 5 above)
Comment (JoeArmani) [reply tests/corpus/scripts/fetchCorpus.mjs]: Developers do not need to set a token. I updated the fetcher so it no longer reads GITHUB_TOKEN or GH_TOKEN. Local runs and CI download the exact pinned commit from the public questionnaire repository, then verify every file against the hashes in the lock file.

File: tests/knownDefects/registry.js (hunk 6 above)
Comment (JoeArmani) [reply tests/knownDefects/registry.js:2]: Yes. This is just to keep tabs on the knownDefect items and make sure they're not forgotten about. I'm in the process of working through a bunch of them with the current work.

**Comment by JoeArmani** [comment]
> Incredibly thorough!! This will be so nice to have and ease developers minds when they make changes in the future. A few small things that Claude pointed out that I felt were worth bringing up.

Thanks! Yeah, I think it'll be super helpful moving forward. Appreciate the code review, and I updated a few things because of it.
